enumerable_method.rb

```diff
@@ -29,180 +30,252 @@ def my_select
     newarray
   end
 
-  def my_all?(param = nil)
-    return false if param.class == Regexp
+  def my_count(item = nil)
+    counter = 0
+    if item.nil?
+      return length unless block_given?
 
-    t = true
-    if block_given?
-      return true unless block_given?
+      each do |x|
+        counter += 1 if yield(x)
+      end
+    else
+      my_each do |y|
+        counter += 1 if y == item
+      end
+    end
+    counter
+  end
 
+  def my_map
+    return to_enum(:my_map) unless block_given?
+
+    newarray = []
+    if self.class == Array
       my_each do |x|
-        t = if_nil_false_return_false yield(x)
-        break if t == false
+        newarray << yield(x) ? true : false
+      end
+    else
+      i = first
+      while i <= last
+        newarray << yield(i) unless yield(i).nil?
+        i += 1
       end
-    elsif param.nil?
+    end
+    newarray
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/PerceivedComplexity
```

Comment:
[Optional]

I would recommend investigating `===` (`CaseEquality`).

This should help test for Class, RegEx and other types of the argument passed at the same time, help reduce the condition branches and method length.